src/main/cli/cli.c

```diff
@@ -5435,11 +5397,6 @@ static void printConfig(char *cmdline, bool doDiff)
 #endif
         }
 
-        if (!(dumpMask & HARDWARE_ONLY)) {
```

Comment:
I think we should keep printing the craft name at the top of the dump, to help users identify what their dumps are for.

Reply:
The bad thing about that is you get it listed twice in a `diff` like:
```
# name
name CraftName

# master
set name = CraftName
```

I could change it to be just a comment at the top rather than a command like:
```
# name CraftName
```

Or we could even let the `name` command still work as a "hidden" feature to ease the transition since people will have saved configs that define it and otherwise all would generate errors until they convert. But of course that doesn't drive anybody to fix their config.

Reply:
I agree that the duplicity isn't great. I think just printing the name as a comment up top will be fine.

Easing the transition sounds like a good thing - we could just keep `name` as a NOOP and stop printing it in `diff` in 4.1 , so users will get a smooth transition, with their 4.0 diffs containing `set name`, and the `name` that it also contains being ignored without throwing an error.

Reply:
Well to was the transition we'd still have to let the `name` command work. Making the command do nothing would just mean that user configs with it would just silently fail and the craft name wouldn't be set (since they probably won't have it as `set name =` in their saved config).

I propose printing it just as a comment at the top now and keeping the `name` command functional to set the name, but have it spit out a warning that the command will be removed in the future. Then in 4.2 we can remove the `name` command.

Reply:
It should work in most cases, assuming that this change goes into 4.1: All of 4.0 will be printing `name` and `set name =` in the `diff`, so only users completely skipping 4.0 with their diffs will be affected.